packages/server/test/regression/issue-949-inferProcedureInput.test.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+// IMPORTANT:
+// needs to be imported from compiled output otherwise we get a false-positive
+import * as trpc from '../../dist/trpc-server.cjs';
+import { expectTypeOf } from 'expect-type';
+import { z } from 'zod';
+
+// https://github.com/trpc/trpc/issues/949
+// https://github.com/trpc/trpc/pull/955
+test('inferProcedureFromInput regression', async () => {
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  type Context = {};
+  const appRouter = trpc.router<Context>().merge(
+    'admin.',
+    trpc
+      .router<Context>()
+      .mutation('testMutation', {
+        input: z.object({
+          in: z.string(),
+        }),
+        resolve: async () => `out` as const,
+      })
+      .query('hello', {
+        input: z.object({
+          in: z.string(),
+        }),
+        resolve() {
+          return 'out' as const;
+        },
+      })
+      .middleware(async ({ next, ctx }) =>
+        next({
+          ctx: {
+            ...ctx,
+            _test: '1',
+          },
+        }),
+      )
+      .query('hello-2', {
+        input: z.object({
+          in: z.string(),
+        }),
+        resolve() {
+          return 'out' as const;
+        },
+      }),
```

Comment:
I think it'd be worth having a test case for a no-input procedure too. Changing from `Procedure` -> `ProcedureWithInput` in the type helper might have made this _only_ work for procedures with input.